source/dynamical_systems/include/dynamical_systems/IDynamicalSystem.hpp

```diff
@@ -0,0 +1,114 @@
+#pragma once
+
+#include <list>
+#include <map>
+#include <memory>
+
+#include "state_representation/space/cartesian/CartesianPose.hpp"
+#include "state_representation/space/cartesian/CartesianState.hpp"
+#include "state_representation/parameters/ParameterInterface.hpp"
+
+/**
+ * @namespace dynamical_systems
+ * @brief Systems of equations relating state variables to their derivatives
+ */
+namespace dynamical_systems {
+
+/**
+ * @class IDynamicalSystem
+ * @brief Abstract class for a dynamical system.
+ * @tparam S Underlying state type of the dynamical system
+ */
+template<class S>
+class IDynamicalSystem {
+public:
+  /**
+   * @brief Empty constructor
+   */
+  IDynamicalSystem();
+
+  /**
+   * @brief Check compatibility between a state and the dynamical system.
+   * @param state The state to check for compatibility
+   * @return True if the state is compatible with the dynamical system
+   */
+  [[nodiscard]] virtual bool is_compatible(const S& state) const;
+
+  /**
+   * @brief Evaluate the value of the dynamical system at a given state.
+   * @param state State at which to perform the evaluation
+   * @return The resulting state (velocity) of the dynamical system
+   */
+  [[nodiscard]] S evaluate(const S& state) const;
```

Comment:
That is a good point. I am not against remove the `const` actually if you think that make more sense. Adding the `const chronotype& dt` could also be a nice solution. Both this and `compute_dynamics` should be treated in the same way I think.

Reply:
I think two methods, a const like this and non-const with the additional argument, will be a nice solution. Since `compute_dynamics` is the function to be overridden, you are correct that it needs to have the equivalent const and non-const variants. This can wait for a future PR though.

Reply:
Okay so nothing to do here for me, at least for now?